File: src/java/org/codehaus/testdox/intellij/actions/BaseAction.java
package org.codehaus.testdox.intellij.actions;

import javax.swing.Icon;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;

abstract class BaseAction extends AnAction {

    protected static final boolean DO_NOT_USE_FROM_TESTDOX_TOOL_WINDOW = false;

    protected final ActionEvents actionEvents = ActionEvents.instance;

    BaseAction(String text, String description, Icon icon) {
        super(text, description, icon);
        getTemplatePresentation().setEnabled(false);
    }

    public void update(AnActionEvent event) {
        event.getPresentation().setEnabled(actionEvents.getTestDoxController(event).hasActiveEditors());
    }
}

File: src/scala/org/codehaus/testdox/intellij/actions/BaseAction.scala
package org.codehaus.testdox.intellij.actions

import javax.swing.Icon

import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent

abstract class BaseAction(text: String, description: String, icon: Icon) extends AnAction(text, description, icon) {

  getTemplatePresentation().setEnabled(false)

  protected val actionEvents = ActionEvents.instance

  override def update(event: AnActionEvent) {
    event.getPresentation().setEnabled(actionEvents.getTestDoxController(event).hasActiveEditors())
  }
}

object BaseAction {
  val DO_NOT_USE_FROM_TESTDOX_TOOL_WINDOW = false
}
